src/qibolab/platforms/platform.py

```diff
@@ -1,6 +1,94 @@
+from dataclasses import asdict, dataclass, replace
+from enum import Enum, auto
+from typing import Optional
+
 from qibo.config import raise_error
 
 from qibolab.platforms.abstract import AbstractPlatform
+from qibolab.result import (
+    AveragedIntegratedResults,
+    AveragedRawWaveformResults,
+    AveragedStateResults,
+    IntegratedResults,
+    RawWaveformResults,
+    StateResults,
+)
+
+
+class AcquisitionType(Enum):
+    """
+    Types of data acquisition from hardware.
+
+    SPECTROSCOPY: Zurich Integration mode for RO frequency sweeps,
+    INTEGRATION: Demodulate and integrate the waveform,
+    RAW: Acquire the waveform as it is,
+    DISCRIMINATION: Demodulate, integrate the waveform and discriminate among states based on the voltages
+
+    """
+
+    RAW = auto()
+    INTEGRATION = auto()
+    DISCRIMINATION = auto()
+    SPECTROSCOPY = auto()
```

Comment:
I don't know what is the standard way to document classes but here you could also add a docstring after each attribute.
```py
RAW = auto()
"""Acquire waverform as it is."""
INTEGRATION = auto()
"""Demodulate and integrate the waveform."""
...
```
and so on.
In this way we could avoid to right all the types for data acquisition more than once.